receiver/receivercreator/observerhandler_test.go

```diff
@@ -18,155 +18,250 @@ import (
+			receiverTemplateConfig: userConfigMap{"int_field": 23456789, "not_endpoint": "not.an.endpoint"},
+			expectedReceiverType:   &nopWithoutEndpointReceiver{},
+			expectedReceiverConfig: &nopWithoutEndpointConfig{
+				IntField:    23456789,
+				NotEndpoint: "not.an.endpoint",
+			},
+		},
+		{
+			name:                   "inherits unsupported endpoint",
+			receiverTemplateID:     component.NewIDWithName("without.endpoint", "some.name"),
+			receiverTemplateConfig: userConfigMap{"endpoint": "unsupported.endpoint"},
+			expectedError:          "failed to load \"without.endpoint/some.name\" template config: 1 error(s) decoding:\n\n* '' has invalid keys: endpoint",
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			cfg := createDefaultConfig().(*Config)
+			rcvrCfg := receiverConfig{
+				id:         test.receiverTemplateID,
+				config:     test.receiverTemplateConfig,
+				endpointID: portEndpoint.ID,
+			}
+			cfg.receiverTemplates = map[string]receiverTemplate{
+				rcvrCfg.id.String(): {
+					receiverConfig:     rcvrCfg,
+					rule:               portRule,
+					Rule:               `type == "port"`,
+					ResourceAttributes: map[string]interface{}{},
+				},
+			}
 
-	set := receivertest.NewNopCreateSettings()
-	set.ID = component.NewIDWithName("name", "1")
-	rcvrCfg := receiverConfig{id: set.ID, config: userConfigMap{"foo": "bar"}, endpointID: portEndpoint.ID}
-	cfg := createDefaultConfig().(*Config)
-	cfg.receiverTemplates = map[string]receiverTemplate{
-		"name/1": {rcvrCfg, "", map[string]interface{}{}, newRuleOrPanic(`type == "port"`)},
-	}
-	handler := &observerHandler{
-		params:                set,
-		config:                cfg,
-		receiversByEndpointID: receiverMap{},
-		runner:                runner,
-	}
+			handler, mr := newObserverHandler(t, cfg)
+			handler.OnAdd([]observer.Endpoint{
+				portEndpoint,
+				unsupportedEndpoint,
+			})
 
-	runner.On(
-		"start",
-		rcvrCfg,
-		userConfigMap{endpointConfigKey: "localhost:1234"},
-		mock.IsType(&resourceEnhancer{}),
-	).Return(&nopWithEndpointReceiver{}, nil)
+			if test.expectedError != "" {
+				assert.Equal(t, 0, handler.receiversByEndpointID.Size())
+				require.Error(t, mr.lastError)
+				require.EqualError(t, mr.lastError, test.expectedError)
+				require.Nil(t, mr.startedComponent)
+				return
+			}
 
-	handler.OnAdd([]observer.Endpoint{
-		portEndpoint,
-		unsupportedEndpoint,
-	})
+			assert.Equal(t, 1, handler.receiversByEndpointID.Size())
+			require.NoError(t, mr.lastError)
+			require.NotNil(t, mr.startedComponent)
 
-	runner.AssertExpectations(t)
-	assert.Equal(t, 1, handler.receiversByEndpointID.Size())
+			var actualConfig component.Config
+			switch v := mr.startedComponent.(type) {
+			case *nopWithEndpointReceiver:
+				require.NotNil(t, v)
+				actualConfig = v.cfg
+			case *nopWithoutEndpointReceiver:
+				require.NotNil(t, v)
+				actualConfig = v.cfg
+			default:
+				t.Fatalf("unexpected startedComponent: %T", v)
```

Comment:
I believe there is `require.Fail(...)` that you can use here instead of calling `t` directly, helps keep the test output consistent.